$NetBSD: patch-configure,v 1.4 2014/07/09 12:48:59 he Exp $

Several fixes:
 - Enable pic on all non-i386 archs
 - Use a more portable endianness-test
 - Improve test of texi2html
 - Disable symbol versioning test
 - Always provide a run-path during linking

--- configure.orig	2014-06-29 01:31:38.000000000 +0000
+++ configure
@@ -2614,6 +2614,11 @@ case $target_os in
         ;;
     netbsd)
         disable symver
+        # NetBSD/macppc *really* needs PIC enabled.  ffmpeg has repeatedly
+        # broken time and time again because PIC support has been removed
+        # from architectures that need it.  So, if you touch this, please
+        # be careful and make sure that your changes work.
+        test "${subarch}" != "x86_32" && enable pic
         oss_indev_extralibs="-lossaudio"
         oss_outdev_extralibs="-lossaudio"
         ;;
@@ -2845,10 +2850,13 @@ check_cc <<EOF && enable attribute_may_a
 union { int x; } __attribute__((may_alias)) x;
 EOF
 
-check_cc <<EOF || die "endian test failed"
-unsigned int endian = 'B' << 24 | 'I' << 16 | 'G' << 8 | 'E';
+check_exec <<EOF || enable bigendian
+int main()
+{
+ long one = 1;
+ return !(*((char *)(&one)));
+}
 EOF
-od -t x1 $TMPO | grep -q '42 *49 *47 *45' && enable bigendian
 
 if enabled alpha; then
 
@@ -3203,7 +3211,7 @@ else
 fi
 enabled sdl && add_cflags $sdl_cflags && add_extralibs $sdl_libs
 
-texi2html -version > /dev/null 2>&1 && enable texi2html || disable texi2html
+which texi2html > /dev/null 2>&1 && enable texi2html || disable texi2html
 makeinfo --version > /dev/null 2>&1 && enable makeinfo  || disable makeinfo
 
 check_header linux/fb.h
@@ -3306,6 +3314,7 @@ check_ldflags -Wl,--warn-common
 check_ldflags -Wl,-rpath-link=libpostproc:libswresample:libswscale:libavfilter:libavdevice:libavformat:libavcodec:libavutil
 test_ldflags -Wl,-Bsymbolic && append SHFLAGS -Wl,-Bsymbolic
 
+if false; then
 echo "X{};" > $TMPV
 if test_ldflags -Wl,--version-script,$TMPV; then
     append SHFLAGS '-Wl,--version-script,\$(SUBDIR)lib\$(NAME).ver'
@@ -3318,6 +3327,7 @@ __asm__(".symver ff_foo,av_foo@VERSION")
 void ff_foo(void) {}
 EOF
 fi
+fi
 
 if [ -n "$optflags" ]; then
     add_cflags $optflags
@@ -3743,7 +3753,7 @@ Version: $version
 Requires: $(enabled shared || echo $requires)
 Requires.private: $(enabled shared && echo $requires)
 Conflicts:
-Libs: -L\${libdir} -l${shortname} $(enabled shared || echo $libs)
+Libs: -L\${libdir} -Wl,-R\${libdir} -l${shortname} $(enabled shared || echo $libs)
 Libs.private: $(enabled shared && echo $libs)
 Cflags: -I\${includedir}
 EOF
